types/reveal/index.d.ts

```diff
@@ -1,241 +1,1701 @@
-// Type definitions for Reveal 4.2.0
+// Type definitions for Reveal 4.3.1
 // Project: https://github.com/hakimel/reveal.js/
 // Definitions by: robertop87 <https://github.com/robertop87>,
 //                 Nava2 <https://github.com/Nava2>,
 //                 JPtenBerge <https://github.com/JPtenBerge>
+//                 Keita Watanabe <https://github.com/kwatanwa17>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-declare const Reveal: RevealStatic;
-
-// Reveal plugins are globals since 4.0 (if not loaded through ES Modules): https://revealjs.com/plugins/
-declare const RevealHighlight: Plugin;
-declare const RevealMarkdown: Plugin;
-declare const RevealSearch: Plugin;
-declare const RevealNotes: Plugin;
-declare const RevealMath: Plugin;
-declare const RevealZoom: Plugin;
-
-interface RevealStatic {
-    initialize: (config?: RevealOptions) => Promise<any>;
-    configure: (diff?: RevealOptions) => void;
-
-    // Navigation
-    slide(indexh: number, indexv?: number, f?: number, o?: number): void;
-    left(): void;
-    right(): void;
-    up(): void;
-    down(): void;
-    prev(): void;
-    next(): void;
-    prevFragment(): boolean;
-    nextFragment(): boolean;
-
-    // Randomize the order of slides
-    shuffle(): void;
-
-    // Toggle presentation states
-    toggleOverview(override?: boolean): void;
-    togglePause(override?: boolean): void;
-    toggleAutoSlide(override?: boolean): void;
-
-    // Retrieves the previous and current slide elements
-    getPreviousSlide(): Element;
-    getCurrentSlide(): Element;
-
-    getIndices(slide?: Element): { h: number; v: number };
-    getProgress(): number;
-    getTotalSlides(): number;
-    availableFragments(): { prev: boolean; next: boolean };
-
-    // Returns the speaker notes for the current slide
-    getSlideNotes(slide?: Element): string;
-
-    // Plugins
-    hasPlugin(name: string): boolean;
-    getPlugin(name: string): Plugin;
-    getPlugins(): { [name: string]: Plugin };
-
-    // States
-    addEventListener(type: string, listener: (event: any) => void, useCapture?: boolean): void;
-    removeEventListener(type: string, listener: (event: any) => void, useCapture?: boolean): void;
-
-    // State Checks
-    isFirstSlide(): boolean;
-    isLastSlide(): boolean;
-    isPaused(): boolean;
-    isOverview(): boolean;
-    isAutoSliding(): boolean;
-
-    // undocumented method
-    layout(): void;
-    addEventListeners(): void;
-    removeEventListeners(): void;
-    getSlide(x: number, y?: number): Element;
-    getScale(): number;
-    getConfig(): RevealOptions;
-    getQueryHash(): any;
-    setState(state: any): void;
-    getState(): any;
-
-    // update slides after dynamic changes
-    sync(): void;
-}
+// declare module as `reveal.js` so that it matched with the packeage name
+declare module 'reveal.js' {
```

Comment:
Rather than declaring this as an ambient module, the package directory (`types/reveal`) should be renamed to reflect the actual package name (i.e., `types/reveal.js`).